src/app/app-error.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, ErrorHandler } from '@angular/core';
+import { ModalService } from './core';
+
+@Injectable()
+export class AppErrorService implements ErrorHandler {
+
+  private defaultHandler: ErrorHandler;
+
+  constructor(private modal: ModalService) {
+    this.defaultHandler = new ErrorHandler(false); // don't re-throw
+  }
+
+  handleError(err) {
+    this.modal.show({
+      title: 'Uncaught ' + (err.name || 'Error'),
+      body: `
+        <p>${err.message}</p>
+        <hr/>
+        <pre style="font-size:12px;line-height:16px;white-space:pre">${err.stack}</pre>
```

Comment:
I was having some problems getting any `modal.component.css` to apply to the actual contents rendered in the modal.  Since angular is doing the whole "emulated shadow dom" thing, all the styles get a bunch of extra attribute-selectors added at runtime, which my `body:` does not include.

Any other ideas?  Could also revisit this when we get even better error handling.  (Prod-vs-dev errors, links to related error data, etc).